### PR TITLE
Add AWS support to mdbd.

### PR DESCRIPTION
**Pull request by keep94**
This adds aws support to mbdb. The driver name is "aws"; the url is the data center like "us-east-1" The aws driver does not support limiting results by datacenter as it currently grabs machine data from a single data center.

### CODE REVIEW

File: cmd/mdbd/main.go [hunk 1]
```diff
@@ -42,6 +42,16 @@ func printUsage() {
...
+		"  aws: Amazon AWS endpoint. url is datacenter like 'us-east-1'.")
+	fmt.Fprintln(os.Stderr,
+		"       This driver requires the file ~/.aws/credentials which")
+	fmt.Fprintln(os.Stderr,
+		"       contains the amazon aws credentials. For additional")
+	fmt.Fprintln(os.Stderr,
+		"       information see:")
+	fmt.Fprintln(os.Stderr,
+		"       http://docs.aws.amazon.com/sdk-for-go/latest/v1/developerguide/sdkforgo-dg.pdf")
 }
```
Comment (rgooch) [inline cmd/mdbd/main.go]: Please sort (i.e. move the AWS section upwards).

File: cmd/mdbd/main.go [hunk 2]
```diff
@@ -70,12 +82,36 @@ func getLogger() *log.Logger {
...
 	driverFunc driverFunc
-	url        string
+	// the url or path of the flat file
+	url string
+}
+
+func (s source) Generate(
+	datacentre string, logger *log.Logger) (*mdb.Mdb, error) {
+	return loadMdb(s.driverFunc, s.url, datacentre, logger)
 }
```
Comment (rgooch) [inline cmd/mdbd/main.go]: I think it might be cleaner to move these types and their methods into a separate file. Perhaps generators.go?

Comment (keep94) [inline cmd/mdbd/main.go]: Done.

File: cmd/mdbd/main.go (hunk 1 above)
Comment (keep94) [inline cmd/mdbd/main.go]: Done.

File: cmd/mdbd/main.go [hunk 3]
```diff
@@ -149,8 +166,13 @@ func main() {
...
+			getSource(
+				flag.Arg(index),
+				flag.Arg(index+1)))
 	}
-	runDaemon(sources, *mdbFile, *hostnameRegex, *datacentre, *fetchInterval,
-		logger, *debug)
+	runDaemon(
+		generators, *mdbFile, *hostnameRegex, *datacentre,
+		*fetchInterval, logger, *debug)
 }
```
Comment (rgooch) [inline cmd/mdbd/main.go]: Please don't reformat like this. Comment applies to both statements.

Comment (keep94) [inline cmd/mdbd/main.go]: Done.

**Pull request merged**